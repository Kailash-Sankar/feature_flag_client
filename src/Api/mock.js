function getPackages() {
  return {
    "0": { id: 0, name: "Custom" },
    "1": { id: 1, name: "Silver" },
    "2": { id: 2, name: "Gold" },
    "3": { id: 3, name: "Diamond" }
  };
}

// hardcoded for demo
function getPackageFeatures() {
  return {
    "0": [],
    "1": {
      ad_slates: {
        id: "ad_slates",
        status: 2
      }
    },
    "2": [],
    "3": []
  };
}

export default {
  getPackages,
  getPackageFeatures
};
