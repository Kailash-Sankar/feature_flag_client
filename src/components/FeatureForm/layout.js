export const formItemLayout = {
  labelCol: {
    xs: { span: 2 },
    sm: { span: 2 }
  },
  wrapperCol: {
    xs: { span: 24 },
    sm: { span: 16 }
  }
};

export const tailFormItemLayout = {
  wrapperCol: {
    xs: {
      span: 2,
      offset: 0
    },
    sm: {
      span: 4,
      offset: 2
    }
  }
};
