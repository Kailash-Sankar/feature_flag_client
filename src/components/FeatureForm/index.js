import React, { useEffect } from "react";
import { Form, Input, Radio, Button } from "antd";
import { formItemLayout, tailFormItemLayout } from "./layout";

function BaseFeatureForm({
  // feature,
  products,
  packages,
  onSubmit,
  reset
}) {
  const [form] = Form.useForm();
  const { resetFields } = form;

  useEffect(() => {
    resetFields();
  }, [reset]);

  // called if validation passes
  function handleSubmit(values) {
    onSubmit(values);
  }

  return (
    <div>
      <Form
        form={form}
        {...formItemLayout}
        onFinish={handleSubmit}
        hideRequiredMark={true}
      >
        <Form.Item
          label="ID"
          name="id"
          rules={[{ required: true, message: "A unique id is required" }]}
        >
          <Input placeholder="Enter unique feature flag id" />
        </Form.Item>

        <Form.Item
          label="Name"
          name="name"
          rules={[{ required: true, message: "Name is required" }]}
        >
          <Input placeholder="Enter a name for the flag" />
        </Form.Item>

        <Form.Item
          label="Description"
          name="description"
          rules={[{ required: true, message: "Description is required" }]}
        >
          <Input
            type="textarea"
            placeholder="Enter a description for the flag"
          />
        </Form.Item>

        <Form.Item
          label="Product"
          name="product"
          rules={[{ required: true, message: "Product is required" }]}
        >
          <Radio.Group>
            {Object.keys(products).map((p) => (
              <Radio key={p} value={p}>
                {products[p].name}
              </Radio>
            ))}
          </Radio.Group>
        </Form.Item>

        <Form.Item
          label="Package"
          name="package"
          rules={[{ required: true, message: "package is required" }]}
        >
          <Radio.Group>
            {Object.keys(packages).map((p) => (
              <Radio key={p} value={p}>
                {packages[p].name}
              </Radio>
            ))}
          </Radio.Group>
        </Form.Item>

        <Form.Item {...tailFormItemLayout}>
          <Button type="primary" htmlType="submit">
            Submit
          </Button>
        </Form.Item>
      </Form>
    </div>
  );
}

export default BaseFeatureForm;
