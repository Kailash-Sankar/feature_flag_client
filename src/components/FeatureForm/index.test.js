import { shallow } from "enzyme";
import React from "react";
import mock from "@mock/data";
import FeatureForm from "./index";

test("feature form", () => {
  const props = {
    products: mock.products,
    packages: mock.packages,
    onSubmit: mock.noop,
    reset: 0
  };
  const wrapper = shallow(<FeatureForm {...props} />);
  expect(wrapper).toMatchSnapshot();
});
