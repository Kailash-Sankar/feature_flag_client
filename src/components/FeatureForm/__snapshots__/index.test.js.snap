// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`feature form 1`] = `
<div>
  <ForwardRef(InternalForm)
    form={
      Object {
        "__INTERNAL__": Object {},
        "getFieldError": [Function],
        "getFieldValue": [Function],
        "getFieldsError": [Function],
        "getFieldsValue": [Function],
        "getInternalHooks": [Function],
        "isFieldTouched": [Function],
        "isFieldValidating": [Function],
        "isFieldsTouched": [Function],
        "isFieldsValidating": [Function],
        "resetFields": [Function],
        "scrollToField": [Function],
        "setFields": [Function],
        "setFieldsValue": [Function],
        "submit": [Function],
        "validateFields": [Function],
      }
    }
    hideRequiredMark={true}
    labelCol={
      Object {
        "sm": Object {
          "span": 2,
        },
        "xs": Object {
          "span": 2,
        },
      }
    }
    onFinish={[Function]}
    wrapperCol={
      Object {
        "sm": Object {
          "span": 16,
        },
        "xs": Object {
          "span": 24,
        },
      }
    }
  >
    <FormItem
      label="ID"
      name="id"
      rules={
        Array [
          Object {
            "message": "A unique id is required",
            "required": true,
          },
        ]
      }
    >
      <Input
        placeholder="Enter unique feature flag id"
        type="text"
      />
    </FormItem>
    <FormItem
      label="Name"
      name="name"
      rules={
        Array [
          Object {
            "message": "Name is required",
            "required": true,
          },
        ]
      }
    >
      <Input
        placeholder="Enter a name for the flag"
        type="text"
      />
    </FormItem>
    <FormItem
      label="Description"
      name="description"
      rules={
        Array [
          Object {
            "message": "Description is required",
            "required": true,
          },
        ]
      }
    >
      <Input
        placeholder="Enter a description for the flag"
        type="textarea"
      />
    </FormItem>
    <FormItem
      label="Product"
      name="product"
      rules={
        Array [
          Object {
            "message": "Product is required",
            "required": true,
          },
        ]
      }
    >
      <RadioGroup
        buttonStyle="outline"
      >
        <Radio
          key="ax"
          type="radio"
          value="ax"
        >
          Ad Insights
        </Radio>
        <Radio
          key="bx"
          type="radio"
          value="bx"
        >
          Experience Insights
        </Radio>
        <Radio
          key="cx"
          type="radio"
          value="cx"
        >
          Content Insights
        </Radio>
        <Radio
          key="dx"
          type="radio"
          value="dx"
        >
          Social Insights
        </Radio>
      </RadioGroup>
    </FormItem>
    <FormItem
      label="Package"
      name="package"
      rules={
        Array [
          Object {
            "message": "package is required",
            "required": true,
          },
        ]
      }
    >
      <RadioGroup
        buttonStyle="outline"
      >
        <Radio
          key="0"
          type="radio"
          value="0"
        >
          Custom
        </Radio>
        <Radio
          key="1"
          type="radio"
          value="1"
        >
          Silver
        </Radio>
        <Radio
          key="2"
          type="radio"
          value="2"
        >
          Gold
        </Radio>
        <Radio
          key="3"
          type="radio"
          value="3"
        >
          Diamond
        </Radio>
      </RadioGroup>
    </FormItem>
    <FormItem
      wrapperCol={
        Object {
          "sm": Object {
            "offset": 2,
            "span": 4,
          },
          "xs": Object {
            "offset": 0,
            "span": 2,
          },
        }
      }
    >
      <Button
        block={false}
        ghost={false}
        htmlType="submit"
        loading={false}
        type="primary"
      >
        Submit
      </Button>
    </FormItem>
  </ForwardRef(InternalForm)>
</div>
`;
